src/main/java/org/opensearch/flowframework/workflow/ToolStep.java

```diff
@@ -126,6 +134,14 @@ private Map<String, String> getToolsParametersMap(
             .map(Map.Entry::getKey)
             .findFirst();
 
+        // Case when connectorId is passed through previousSteps and not present already in parameters
+        if (previousNodeConnector.isPresent() && !parametersMap.containsKey(CONNECTOR_ID)) {
+            WorkflowData previousNodeOutput = outputs.get(previousNodeConnector.get());
+            if (previousNodeOutput != null && previousNodeOutput.getContent().containsKey(CONNECTOR_ID)) {
+                parametersMap.put(CONNECTOR_ID, previousNodeOutput.getContent().get(CONNECTOR_ID).toString());
+            }
+        }
```

Comment:
Can be done the same for this lines as well?

Reply:
Updated.